frontend/src/lib/components/SelectBox.tsx

```diff
@@ -0,0 +1,248 @@
+    name: JSX.Element | string
+}
+
+export interface SelectedItem {
+    name: string
+    key: string
+    action?: ActionType
+    event?: string
+    volume?: number
+    usage_count?: number
+    category?: string
+}
+
+const scrollUpIntoView = (key: string): void => {
+    const searchList = document.querySelector('.search-list')
+    const item = document.querySelector('.action-filter-dropdown [datakey="' + key + '"]')
+    const diff = item?.getBoundingClientRect().top - searchList?.getBoundingClientRect().top
+    if (diff - 30 < 0) searchList.scrollTop = searchList.scrollTop + diff - 30
+}
+const scrollDownIntoView = (key: string): void => {
+    const searchList = document.querySelector('.search-list')
+    const item = document.querySelector('.action-filter-dropdown [datakey="' + key + '"]')
+    const diff = item?.getBoundingClientRect().top - searchList?.getBoundingClientRect().bottom
+    if (diff + 30 > 0) searchList.scrollTop = searchList.scrollTop + diff + 30
+}
+
+export const actionFilterDropdownLogic = kea({
+    actions: {
+        setSelectedItem: (item: SelectedItem) => ({ item }),
+        setSearch: (search: string) => ({ search }),
+        clickSelectedItem: (item: SelectedItem) => ({ item }),
+        setBlockMouseOver: (block: boolean) => ({ block }),
+        onKeyDown: (e) => ({ e }),
+    },
+    reducers: ({ props }) => ({
+        selectedItem: [
+            false,
+            {
+                setSelectedItem: (_, { item }: { item: SelectedItem }) => item,
+            },
+        ],
+        blockMouseOver: [
+            false,
+            {
+                setBlockMouseOver: (_, { block }: { block: boolean }) => block,
+            },
+        ],
+        search: [
+            false,
+            {
+                setSearch: (_, { search }: { search: string }) => search,
+            },
+        ],
+        RenderInfo: [
+            false,
+            {
+                setSelectedItem: (_, { item }: { item: SelectedItem }) => {
+                    return (
+                        props.items.filter((i) => i.dataSource.filter((i) => i.key === item.key).length > 0)[0]
+                            ?.renderInfo || false
+                    )
+                },
+            },
+        ],
+    }),
+    listeners: ({ props, values, actions }) => ({
+        clickSelectedItem: ({ item }: { item: SelectedItem }) => {
+            if (item.event) {
+                props.updateFilter(EntityTypes.EVENTS, item.event, item.event)
+            } else {
+                props.updateFilter(EntityTypes.ACTIONS, item.action.id, item.action.name)
+            }
+        },
+        setBlockMouseOver: ({ block }) => {
+            if (block) setTimeout(() => actions.setBlockMouseOver(false), 200)
+        },
+        onKeyDown: ({ e }) => {
+            let allSources = props.items.map((item) => item.dataSource).flat()
+            allSources = !values.search
```

Comment:
Note this is repeated twice and is not the most obvious code. I think it's worth extracting a function and giving it a good name:

```
function searchSourcesByName(sources: T, search: string): SomeType
```